core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -1129,26 +1113,41 @@ func (c *coordinator) emitReportWillBeTransmittedMetrics(
 	promCallbacksInReport.WithLabelValues(c.labelValues()...).Observe(float64(numCallbacks))
 }
 
-func filterName(beaconAddress, coordinatorAddress, dkgAddress common.Address) string {
-	return logpoller.FilterName("VRF Coordinator", beaconAddress, coordinatorAddress, dkgAddress)
-}
-
-func FilterNamesFromSpec(spec *job.OCR2OracleSpec) (names []string, err error) {
+func FiltersFromSpec(spec *job.OCR2OracleSpec) (filters []logpoller.Filter, err error) {
 	var cfg ocr2vrfconfig.PluginConfig
-	var beaconAddress, coordinatorAddress, dkgAddress ethkey.EIP55Address
+	var beacon, dkg ethkey.EIP55Address
 
 	if err = json.Unmarshal(spec.PluginConfig.Bytes(), &cfg); err != nil {
-		err = errors.Wrap(err, "failed to unmarshal ocr2vrf plugin config")
+		err = pkgerrors.Wrap(err, "failed to unmarshal ocr2vrf plugin config")
 		return nil, err
 	}
 
-	if beaconAddress, err = ethkey.NewEIP55Address(spec.ContractID); err == nil {
-		if coordinatorAddress, err = ethkey.NewEIP55Address(cfg.VRFCoordinatorAddress); err == nil {
-			if dkgAddress, err = ethkey.NewEIP55Address(cfg.DKGContractAddress); err == nil {
-				return []string{filterName(beaconAddress.Address(), coordinatorAddress.Address(), dkgAddress.Address())}, nil
-			}
-		}
+	beacon, err = ethkey.NewEIP55Address(spec.ContractID)
```

Comment:
This error while parsing the beacon address is joined together with 2 other errors later (the ones from parsing coordinator address & DKG address), and all of them returned together as  a single multi-error.  If any of them are non-nil then the overall error returned will be non-nil so that all of them will be logged as errors.

Reply:
Actually, there is a problem with it though:  if any of them are non-nil then there will be a nil-pointer exception while constructing the return value.  I'll have to fix that, thanks!

Reply:
Apparently it's okay somehow, there is a test for it that calls it with invalid addresses, forcing these errors... and the test passes rather than resulting in a panic.  I guess that means the `Address()` method checks for the nil condition already... nevertheless, adding a nil check to be safer.